pageserver/src/walredo.rs

```diff
@@ -838,6 +894,14 @@ impl PostgresRedoProcess {
     }
 }
 
+impl Drop for PostgresRedoProcess {
+    fn drop(&mut self) {
+        if !self.called_kill {
+            error!(tenant_id=%self.tenant_id, pid = %self.child.id(), "dropping PostgresRedoProcess that wasn't killed, likely causing defunct postgres process");
+        }
```

Comment:
We could do the kill + try_wait.

Reply:
We'll do the fix in a separate change, once we know this is the root cause.

Reply:
Recording some of the discussion we had; it's unlikely that if a kill was really needed the `try_wait` would "see" it right away, would have to have some sleep+try_wait loop, or just be able to wait for the pid non-blockingly like with tokio.

But yeah resolving this.